WordPress/WordPressTest/LikeUserHelperTests.swift

```diff
@@ -0,0 +1,74 @@
+    func createTestRemoteUserDictionary(withPreferredBlog hasPreferredBlog: Bool) -> [String: Any] {
+        var remoteUserDictionary: [String: Any] = [
+            "ID": 15,
+            "login": "testlogin",
+            "name": "testname",
+            "site_ID": 20,
+            "avatar_URL": "wordpress.org/test2",
+            "bio": "testbio",
+            "date_liked": "2021-11-24T04:02:42+0000",
+        ]
+
+        if hasPreferredBlog {
+            remoteUserDictionary["preferred_blog"] = [
+                "id": 1,
+                "url": "wordpress.org/test1",
+                "name": "testblog",
+                "icon": [
+                    "img": "someimage.jpg",
+                ]
+            ]
+        }
+
+        return remoteUserDictionary
+    }
+
+    func testNewLikeUserWithPreferredBlog() {
+        let completionExpectation = expectation(description: "We expect the context to save successfully")
+        let context = TestContextManager.sharedInstance().mainContext
+
+        let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
+        let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
+        let likeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        XCTAssertNotNil(likeUser)
+
+        TestContextManager.sharedInstance().save(context) {
+            completionExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 0.1)
+    }
+
+    func testUpdatingExistingUserToRemovePreferredBlog() {
+        let completionExpectation = expectation(description: "We expect the context to save successfully")
+        let context = TestContextManager.sharedInstance().newDerivedContext()
+
+        // First we create the pre-existing user, so we can later modify it to remove the preferred blog
+
+        let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
+        let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
+        let existingLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        let existingPreferredBlog = existingLikeUser.preferredBlog
+        XCTAssertNotNil(existingLikeUser)
+
+        TestContextManager.sharedInstance().save(context)
+
+        //let updateContext = TestContextManager.sharedInstance().newDerivedContext()
+
+        // Then we remove the preferred blog from the remote user, so we can save it again and make sure
+        // the preferred blog deletion works fine.
+        remoteUser.preferredBlog = nil
+        let updatedLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        XCTAssertNotNil(updatedLikeUser)
+        XCTAssertTrue(existingPreferredBlog?.isDeleted ?? false)
+
+        TestContextManager.sharedInstance().save(context) {
+            completionExpectation.fulfill()
+        }
```

Comment:
As suggested in Slack: it would be better for this Unit Test to properly fail, as opposed to crashing the test runner (as per the `[NSException raise]` that got triggered on failure from `ContextManager+ErrorHandling.swift:69`).

One way to do this would be to have the `TestContextManager` override `handleSaveError(_:in:)` to catch the errors – instead of raising an exception (that would make the app or test completely crash).
Another (maybe easier and better?) way would be to have your test catch the raised exception using `XCTAssertThrows`.

Reply:
Addressed.